File: repetitivas/ej2.py
# -*- coding: utf-8 -*-

#Realizar un algoritmo que pida números (se pedirá por teclado la cantidad de números a introducir). 
#El programa debe informar de cuantos números introducidos son mayores que 0, menores que 0 e iguales a 0.

#almacenar cuantos números introduciremos
#almacenar numero
#si es mayor que cero
#incrementa contador de mayores que 0
#si es menor que cero
#incrementa contador de menores que 0
#si es 0 
#incrementa contador de iguales que 0
#mostrar contadores

print("::-... Comparador de valores con 0 ...-::")
contadorNumeros = 0
cantidadNumeros = int(input("¿Cuantos números vas a introducir> ?"))
menor = 0
mayor = 0
cero = 0

while cantidadNumeros > contadorNumeros:
    numero = int(input("Dime un número "))
    if(numero<0):
        menor = menor + 1
    elif(numero>0):
        mayor = mayor + 1
    else:
        cero = cero + 1
    contadorNumeros = contadorNumeros + 1
			
print(f"Finalizamos el recuento con {mayor} números mayores que cero, {menor} menores y {cero} igual a cero.")

File: Java/Repetitivas/ej2.java
import java.util.Scanner; //Importar la clase Scanner

/**
 * Realizar un algoritmo que pida números 
 * (se pedirá por teclado la cantidad de números a introducir). 
 * El programa debe informar de cuantos números introducidos son 
 * mayores que 0, menores que 0 e iguales a 0.
 * 
 * @author Alberto Miguel Martinez Jimenez
 */
/**
 * Algoritmo
 * Almacenar cantidad de nùmeros
 * Almacenar número
 * Si es mayor que 0 incrementar contadorPositivo
 * Si es menor a 0 incrementar contadorNegatico
 * Si es igual a 0 incrementar contadorCero
 * Mostrar los tres contadores
 */

public class ej2 {

	public static void main(String[] args) {
		int cantidadNumeros;
		double numero;
		int positivos = 0;
		int cero = 0;
		int negativos = 0;
		Scanner scanner = new Scanner(System.in);//Intanciar el objeto Scanner
		
		System.out.println("---Introducir números---");
		System.out.println("¿Cuantos números quieres introducir?> ");
		cantidadNumeros = scanner.nextInt();
		
		for(int i=0;i<cantidadNumeros;i++) {
			System.out.println("Introduce un numero> ");
			numero = scanner.nextDouble();
			if(numero>0) {
				positivos++;
			}else if(numero<0) {
				negativos++;
			}else {
				cero++;
			}
		}
		System.out.println("Has introducido "+positivos+" positivos, "+negativos+" negativos "+" y "+cero+" ceros");
	}

}
